### PR TITLE
Performance Tests: Separate page setup from test

### PR DESCRIPTION
**Pull request by WunderBart**
## What?

This PR introduces a distinct separation between browser contexts used for setting up the test page and those used for running the actual tests. By doing this, we aim to boost metric stability, ensuring that any potential side effects from the setup phase don't impact the test metrics. By having a distinct testing phase, we can also iterate over the entire phase rather than within it (e.g., loading tests). This guarantees that every new page starts with identical initial parameters and no background pages remain open.

Additionally, this modification enables us to make full use of [Playwright fixtures](https://playwright.dev/docs/test-fixtures). These fixtures enhance consistency and minimize code duplication.

> **NOTE**
> The Metrics fixture in this PR will conflict with #52993. It was extracted in a way so that the resolution would be fairly straightforward, though. 🤞 /cc @swissspidy 

## Testing Instructions
Performance tests should pass.

### CODE REVIEW

File: test/performance/specs/front-end-classic-theme.spec.js
Comment (WunderBart) [inline test/performance/specs/front-end-classic-theme.spec.js]: Just making this title consistent with the [other one](https://github.com/WordPress/gutenberg/blob/2c2f572e36ef417c1dedd01222c94a12615dcaa2/test/performance/specs/front-end-block-theme.spec.js#L31).

**Comment by WunderBart** [comment]
Running against the base branch in https://github.com/WordPress/gutenberg/actions/runs/5902472001/job/16010572760

File: test/performance/fixtures/metrics.js
Comment (WunderBart) [inline test/performance/fixtures/metrics.js:1]: Methods in this file are extracted from [utils.js](https://github.com/WordPress/gutenberg/pull/53808/files#diff-48cc25bf23f094d5f93d72f23c413a50e86e73eff6314f88a92e4a8f9946c211)

File: test/performance/fixtures/perf-utils.js
Comment (WunderBart) [inline test/performance/fixtures/perf-utils.js:1]: Methods in this file are extracted from [utils.js](https://github.com/WordPress/gutenberg/pull/53808/files#diff-48cc25bf23f094d5f93d72f23c413a50e86e73eff6314f88a92e4a8f9946c211)

File: test/performance/fixtures/metrics.js
Comment (swissspidy) [inline test/performance/fixtures/metrics.js]: Block selector? AFAICT this is not tied to blocks here, so perhaps remove the comment?

**Comment by swissspidy** [comment]
Thanks for the ping! On vacation right now, so only glanced over the code on my phone, but makes sense to put the fixtures there. I hope with #52993 we can then move the more reusable utils to the shared package

**Review by dmsnell — COMMENTED** [review]
Seems like a fine change to me, but I didn't scrutinize it too deeply. cc: @sgomes since you were involved in this effort before.

Thanks as always @WunderBart!

File: test/performance/fixtures/metrics.js
Comment (WunderBart) [reply test/performance/fixtures/metrics.js]: Nice catch - that's a leftover from copypasting it from the tests. Thanks!

Comment (WunderBart) [reply test/performance/fixtures/metrics.js]: Addressed in https://github.com/WordPress/gutenberg/pull/53808/commits/7000dec29793b8db426b80a4a70bdcdd823aa469

**Comment by swissspidy** [comment]
FYI #52993 is now merged.

----

It's a bit confusing that `startTracing` & `stopTracing` are in `PerfUtils` and utils like `getTypingEventDurations` (which need a trace) are in the `Metrics` class. Maybe these should all be in `PerfUtils`?

**Comment by WunderBart** [comment]
> FYI #52993 is now merged.

Thanks for the heads up!

> It's a bit confusing that `startTracing` & `stopTracing` are in `PerfUtils` and utils like `getTypingEventDurations` (which need a trace) are in the `Metrics` class. Maybe these should all be in `PerfUtils`?

Right, it does make more sense to keep them with `PerfUtils` at this point. 👍 

_edit:_

OTOH, wouldn't it make more sense to move `startTracing` and `stopTracing` to `Metrics` instead? `PerfUtils` are mostly about extra helpers for the editor, and all the `*Durations` getters fit `Metrics` better, IMO.

File: packages/e2e-test-utils-playwright/src/lighthouse/index.ts
Comment (WunderBart) [inline packages/e2e-test-utils-playwright/src/lighthouse/index.ts:16]: Just making it consistent with how we define other fixtures!

**Comment by WunderBart** [comment]
@swissspidy I moved the event tracing to Metrics in https://github.com/WordPress/gutenberg/pull/53808/commits/1b0074193fe13ea72231bfbe1cf9fc43c36966f4. Let me know if that looks OK to you! Also, cc @kevin940726 here for insight on extending these fixtures. 🙏

File: packages/e2e-test-utils-playwright/src/metrics/index.ts
Comment (kevin940726) [inline packages/e2e-test-utils-playwright/src/metrics/index.ts]: I'm curious to know whether playwright automatically infers the return type of `evaluate` here so that we don't have to explicitly cast it as it might be out-of-sync. 🤔

File: packages/e2e-test-utils-playwright/src/metrics/index.ts
Comment (kevin940726) [inline packages/e2e-test-utils-playwright/src/metrics/index.ts]: Nit: Instead of casting as number, how about we just use `!`?

```suggestion
			)!.startTime;
```

File: test/performance/fixtures/perf-utils.js
Comment (kevin940726) [inline test/performance/fixtures/perf-utils.js:1]: Nit: Let's refactor this file into typescript if the tooling allows it?

**Review by kevin940726 — APPROVED** [review]
LGTM as long as it works! 😆 
We can always iterate if needed 👍

File: packages/e2e-test-utils-playwright/src/metrics/index.ts
Comment (WunderBart) [reply packages/e2e-test-utils-playwright/src/metrics/index.ts]: Good call. Looks like there were more places where we cast unnecessarily. Addressed in https://github.com/WordPress/gutenberg/pull/53808/commits/33a78ebaa7521edca5f5b68c5cda0db14ed22210

File: packages/e2e-test-utils-playwright/src/metrics/index.ts
Comment (WunderBart) [reply packages/e2e-test-utils-playwright/src/metrics/index.ts]: Updated in https://github.com/WordPress/gutenberg/pull/53808/commits/feb23d53a20f9b7986339d636ba323d90c15e442

File: test/performance/fixtures/perf-utils.js
Comment (WunderBart) [reply test/performance/fixtures/perf-utils.js:1]: Refactored in https://github.com/WordPress/gutenberg/pull/53808/commits/5c67b2d621f9822f474c47190e1f59b094d906a0

File: packages/e2e-test-utils-playwright/src/metrics/index.ts
Comment (WunderBart) [reply packages/e2e-test-utils-playwright/src/metrics/index.ts]: Removed a little bit too much, reverted some lines in https://github.com/WordPress/gutenberg/pull/53808/commits/71a1d581100e0348b5323fdd615c5095acaf3461 😅

**Comment by WunderBart** [comment]
The base branch comparison is failing after merging this PR. I'm currently investigating it. If I can't find a fix within the next hour, I will revert. 

cc @youknowriad @kevin940726 

_edit:_

This should fix the issue: https://github.com/WordPress/gutenberg/pull/54690

_edit 2:_

[It did](https://github.com/WordPress/gutenberg/commit/6f653f309f2cbc61b89f07dd2e670d74013134f6). 😌

**Pull request merged**
